core/src/Streamly/Internal/Data/Stream/Bottom.hs

```diff
@@ -174,33 +180,25 @@ relTimesWith = fmap snd . timesWith
 -- Example, to continue folding a list of streams on the same sum fold:
 --
 -- >>> streams = [Stream.fromList [1..5], Stream.fromList [6..10]]
--- >>> f = Prelude.foldl Stream.buildl Fold.sum streams
--- >>> Fold.extractM f
+-- >>> f = Prelude.foldl Stream.foldAddLazy Fold.sum streams
+-- >>> Stream.fold f Stream.nil
 -- 55
 --
-{-# INLINE buildl #-}
-buildl :: Monad m => Fold m a b -> Stream m a -> Fold m a b
-buildl f s = D.foldContinue f $ toStreamD s
+{-# INLINE foldAddLazy #-}
+foldAddLazy :: Monad m => Fold m a b -> Stream m a -> Fold m a b
+foldAddLazy f s = D.foldContinue f $ toStreamD s
 
--- | Append a stream to a fold strictly to build an accumulator incrementally.
---
--- Definitions:
---
--- >>> build f = Stream.fold (Fold.duplicate f)
--- >>> build f = Stream.buildl f >=> Fold.reduce
---
--- Example:
---
--- >>> :{
--- do
---  sum1 <- Stream.build Fold.sum (Stream.enumerateFromTo 1 10)
---  sum2 <- Stream.build sum1 (Stream.enumerateFromTo 11 20)
---  Stream.fold sum2 (Stream.enumerateFromTo 21 30)
--- :}
--- 465
+-- >>> foldAdd f = Stream.foldAddLazy f >=> Fold.reduce
+
+-- |
+-- >>> foldAdd = flip Fold.addStream
 --
-build :: Monad m => Fold m a b -> Stream m a -> m (Fold m a b)
-build f = fold (Fold.duplicate f)
+foldAdd :: Monad m => Fold m a b -> Stream m a -> m (Fold m a b)
```

Comment:
Is this `flip addStream`?
Can we discuss these names? 
Names of all the builders and family?

Reply:
Yes, and yes. `foldAdd` is not exposed. The main operation is `addStream`.